addons/mail/models/mail_shortcode.py

```diff
@@ -20,3 +20,38 @@ class MailShortcode(models.Model):
         help="Content that will automatically replace the shortcut of your choosing. This content can still be adapted before sending your message.")
     description = fields.Char('Description')
     last_used = fields.Datetime('Last Used', help="Last time this shortcode was used")
+    group_ids = fields.Many2many("res.groups", string="Authorized Groups", groups="base.group_system")
+    user_ids = fields.Many2many("res.users", string="Authorized Users", groups="base.group_system")
+    is_shared = fields.Boolean(string="Determines if the shortcode is currently shared with other users", compute="_compute_is_shared", store=True)
+
+    @api.depends("group_ids", "user_ids")
+    def _compute_is_shared(self):
+        for shortcode in self:
+            shortcode.is_shared = True if shortcode.group_ids or shortcode.user_ids else False
+
+    def create(self, vals):
+        res = super().create(vals)
+        payload = res._read_format(["id", "source", "substitution"])
+        notifications = [(self.env.user.partner_id, "mail.shortcode/insert", payload)]
+        for group in res.group_ids:
+            notifications.append((group, "mail.shortcode/insert", payload))
+        self.env["bus.bus"]._sendmany(notifications)
+        return res
+
+    def write(self, vals):
+        res = super().write(vals)
+        if res:
+            payload = self._read_format(["id", "source", "substitution"])
+            notifications = [(self.env.user.partner_id, "mail.shortcode/insert", payload)]
+            for group in self.group_ids:
+                notifications.append((group, "mail.shortcode/insert", payload))
+            self.env["bus.bus"]._sendmany(notifications)
+        return res
+
+    def unlink(self):
+        payload = self._read_format(["id"])
+        notifications = [(self.env.user.partner_id, "mail.shortcode/delete", payload)]
+        for group in self.group_ids:
+            notifications.append((group, "mail.shortcode/delete", payload))
+        self.env["bus.bus"]._sendmany(notifications)
```

Comment:
`mail.shortcode/insert` can be turned into `mail.record/insert` with this formatted data:
```js
{
      CannedResponse: {
            id,
            source,
            substitution,
      },
}
```


Ok to keep `mail.shortcode/delete` for the time being, probably in the future we'll plan to introduce `mail.record/delete` or adapt formatted data so `mail.record/insert` can delete targeted records easily (probably the notification will be renamed, e.g. `mail.record/data` or something, who knows)